packages/core/src/repositories/documentVersionsRepository.ts

```diff
@@ -0,0 +1,153 @@
+import { NotFoundError, Result } from '$core/lib'
+import { commits, documentVersions, projects } from '$core/schema'
+import { and, eq, isNotNull, lte, max } from 'drizzle-orm'
+
+import { CommitsRepository } from './commitsRepository'
+import Repository from './repository'
+
+export class DocumentVersionsRepository extends Repository {
+  get scope() {
+    return this.db
+      .select()
+      .from(documentVersions)
+      .innerJoin(projects, eq(projects.workspaceId, this.workspaceId))
+      .innerJoin(commits, eq(commits.projectId, projects.id))
+      .where(eq(documentVersions.commitId, commits.id))
+      .as('documentVersionsScope')
+  }
+
+  async getDocumentById(documentId: number) {
+    const res = await this.db
+      .select()
+      .from(this.scope)
+      .where(eq(documentVersions.id, documentId))
+
+    // NOTE: I hate this
+    const document = res[0]?.document_versions
```

Comment:
This can be solved. Drizzle has a `getColumns` method

``` javascript
const columns = getColumns(db.documentVersions)
const res = await this.db.select(columns)...
```
And now you have a nice array of document versions

```